csrc/preseg_passes/propagate_shardings.cpp

```diff
@@ -353,25 +414,11 @@ void PropagateShardingsPass::runPass(Fusion* fusion) {
       std::unordered_set<ParallelType> selected_parallel_types =
           getParallelTypesToPropagate(outputs_without_mesh);
 
-      // This restricts the transform propagation to only the relevant DID axis.
-      int64_t did_pos =
-          selectiveReorderDIDToFront(ref_input, selected_parallel_types);
-
-      if (auto* view_op = dynamic_cast<ViewOp*>(expr)) {
-        // Propagation of reshape will return how many DID axis were propagated.
-        // They are reordered behind non-propagated DID axis
-        did_pos = shardViewOp(view_op, did_pos);
-      }
-
-      // Propagate the DID loop split to the outputs without mesh.
       propagateDIDTransform(
           /*ref=*/ref_input,
           /*tvs=*/outputs_without_mesh,
-          /*did_pos=*/did_pos,
+          /*selected_parallel_types=*/selected_parallel_types,
```

Comment:
```suggestion
          selected_parallel_types,
```
C++ checks types and selected_parallel_types has a unique type from other arguments.